### ISSUE TITLE
[3.1] Testing with Form & Table Generators from Orchestra/Html

### ISSUE THREAD

**Issue by stevebauman** [orchestral/testing#2]
I'm not sure if I should place this issue here or on `orchestra/html`, but when I run phpunit and call a route with a Table generator instance on it, I receive an `InvalidArgumentException` exception due to the configuration in `resources/packages/orchestra/html/table.php` configuration not being loaded during testing. I've tried overriding it by overriding the `getEnvironmentSetUp($app)` in my test case:

``` php
protected function getEnvironmentSetUp($app)
{
    $app['config']->set('orchestra/html::table', [
        'empty' => 'There are no records to display.',
        'view' => 'components.table',
    ]);

    // Also tried setting the key `orchestra/html.table`
}
```

My test:

``` php
public function test_can_access_issues_with_auth()
{
    $user = factory(User::class)->create();

    $this->actingAs($user);

    $this->visit(route('issues.index'));
}
```

(Partial stack)

``` php
Caused by
exception 'InvalidArgumentException' with message 'View [] not found.' in C:\Program Files (x86)\Ampps\www\www.it-track.dev\vendor\laravel\framework\src\Illuminate\View\FileViewFinder.php:137
Stack trace:
#0 C:\Program Files (x86)\Ampps\www\www.it-track.dev\vendor\laravel\framework\src\Illuminate\View\FileViewFinder.php(79): Illuminate\View\FileViewFinder->findInPaths('', Array)
#1 C:\Program Files (x86)\Ampps\www\www.it-track.dev\vendor\laravel\framework\src\Illuminate\View\Factory.php(151): Illuminate\View\FileViewFinder->find('')
#2 C:\Program Files (x86)\Ampps\www\www.it-track.dev\vendor\orchestra\html\src\Table\TableBuilder.php(52): Illuminate\View\Factory->make(NULL)
```

This issue disappears completely if I specify the layout manually inside the `TableGrid` instance:

``` php
$table->layout('components.table');
```

Would you happen to have any suggestions? Thanks in advance!

EDIT:

Also, I'm not sure why, but even when setting the environment configuration, this doesn't take effect in the `Table\Grid->initiate(Repository $config)` method: https://github.com/orchestral/html/blob/master/src/Table/Grid.php#L94

``` php
// Line 96 in orchestral/html/src/Table/Grid.php
$namespace = ($config instanceof PackageRepository ? 'orchestra/html::table' : 'orchestra.table');

dd($config->get('orchestra/html::table')); // Returns empty array

foreach ($config->get($namespace, []) as $key => $value) {
    if (property_exists($this, $key)) {
        $this->{$key} = $value;
    }
}
```

**Comment by stevebauman** [comment]
Apologies, not sure if this is still an issue or not, but I've fixed this by setting the configuration inside the `setUp()` method in my base test case:

``` php
public function setUp()
{
    $this->app['config']->set('orchestra/html::table', [
        'empty' => 'There are no records to display.',
        'view' => 'components.table',
    ]);
}
```

This also occurs with the HTML `Form\Grid`, I just need to include its configuration inside the `setUp()` method as well.

**Comment by crynobone** [comment]
The only reason this could happen is that the IoC bind before `boot()`.

**Comment by crynobone** [comment]
Fixed via orchestral/html@db4a1bb
